### PR TITLE
Predict_proba doesn't produce probabilities

### PR DESCRIPTION
**Pull request by amueller**
I had problems doing evaluations with the current code, because it seemed like probabilities always sum to 16, as you can check with
```python
from hyperfast import HyperFastClassifier
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris
X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y)
hf = HyperFastClassifier().fit(X_train, y_train)
hf.predict_proba(X_test).sum(axis=1)
```
I assume the line I changed in this PR is responsible? I didn't actually test the change yet, though.

### CODE REVIEW

**Comment by LennartPurucker** [comment]
This seems to be a consequence of the new default values; this works as expected:

```python
from hyperfast import HyperFastClassifier
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris
X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y)

old_defaults = dict(n_ensemble=1, optimization=None, optimize_steps=0)
hf = HyperFastClassifier(**old_clf_args).fit(X_train, y_train)
print(hf.predict_proba(X_test).sum(axis=1))
```

As Andreas spotted, likely a result of ensembling: 

```python
from hyperfast import HyperFastClassifier
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_iris
X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y)

old_clf_args = dict(n_ensemble=1)
hf = HyperFastClassifier(**old_clf_args).fit(X_train, y_train)
print(hf.predict_proba(X_test).sum(axis=1))
```

And I can confirm that @amueller's proposed fix works.

**Comment by davidbonet** [comment]
Missed the probability normalization for ensembling in the cleaned scikit-learn-like interface. This works as well, thanks!
